files/bootstrap.sh

```diff
@@ -343,8 +343,14 @@ if [[ "$MACHINE" != "x86_64" && "$MACHINE" != "aarch64" ]]; then
     exit 1
 fi
 
+AWS_ECR_SUBDOMAIN="ecr"
+# If FIPS is enabled on the machine, use the FIPS endpoint for AWS ECR.
+if grep "fips=1" /etc/default/grub; then
+    AWS_ECR_SUBDOMAIN="ecr-fips"
+fi
+
 PAUSE_CONTAINER_ACCOUNT=$(get_pause_container_account_for_region "${AWS_DEFAULT_REGION}")
-PAUSE_CONTAINER_IMAGE=${PAUSE_CONTAINER_IMAGE:-$PAUSE_CONTAINER_ACCOUNT.dkr.ecr.$AWS_DEFAULT_REGION.$AWS_SERVICES_DOMAIN/eks/pause}
+PAUSE_CONTAINER_IMAGE=${PAUSE_CONTAINER_IMAGE:-$PAUSE_CONTAINER_ACCOUNT.dkr.$AWS_ECR_SUBDOMAIN.$AWS_DEFAULT_REGION.$AWS_SERVICES_DOMAIN/eks/pause}
```

Comment:
nice!

Reply:
Correct me if I'm wrong, but I don't believe that value is available in the context of `bootstrap.sh`. `enable_fips_mode` is known at pack/build/make time and not when the node is joining the cluster. We could effectively pack the image with the value (this is what's being done here albeit indirectly), or autodetect whether or not FIPS is enabled (maybe from kernel parameters?)

Reply:
Yep, you're right; I was confused about where this line was. I think this detection method is fine.